docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,410 @@
+---
+title: Configuration Migration Guide
+eleventyNavigation:
+    key: migration guide
+    parent: configure
+    title: Configuration Migration Guide
+    order: 8
+---
+
+This guide provides an overview of how you can migrate your ESLint configuration file from the eslintrc format (typically configured in `.eslintrc.js` or `.eslintrc.json` files) to the new flat config format (typically configured in an `eslint.config.js` file).
+
+To learn more about the flat config format, refer to [this blog post](https://eslint.org/blog/2022/08/new-config-system-part-2/).
+
+For reference information on these configuration formats, refer to the following documentation:
+
+* [eslintrc configuration files](configuration-files)
+* [flat configuration files](configuration-files-new)
+
+## Start Using Flat Config Files
+
+Starting with ESLint v9.0.0, the flat config file format will be the default configuration file format. Once ESLint v9.0.0 is released, you can start using the flat config file format without any additional configuration.
+
+To use flat config with ESLint v8, place a `eslint.config.js` file in the root of your project **or** set the `ESLINT_USE_FLAT_CONFIG` environment variable to `true`.
+
+## Things That Haven’t Changed between Configuration File Formats
+
+While the configuration file format has changed from eslintrc to flat config, the following has stayed the same:
+
+* Syntax for configuring rules
+* Syntax for configuring processors
+* The CLI, except for the flag changes noted in [CLI Flag Changes](#cli-flag-changes).
+* Global variables are configured the same way, but on a different property (see [Configuring Language Options](#configuring-language-options)).
+
+## Key Differences between Configuration Formats
+
+A few of the most notable differences between the eslintrc and flat config formats are the following:
+
+### Importing Plugins and Custom Parsers
```

Comment:
To me, the key difference is that in eslintrc files, plugins are specified by a string and loaded from a package by ESLint itself. On the other hand, with the flat config, it's the config file that is responsible for loading a plugin as an object (with `require()` or `import()`), and it's this object that is passed to ESLint.
Ultimately, with the flat config, the plugin doesn't have to be loaded from a package; it can be created by any programmatic means.

Reply:
rephrased here [ff8aa3a](https://github.com/eslint/eslint/pull/17230/commits/ff8aa3a2de96f3a1cdd7a896ced93b7c69962f55)